Destination/RotatableStreamDestination.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Zenstruck\BackupBundle\Destination;
+
+use Psr\Log\LoggerInterface;
+use Zenstruck\BackupBundle\Rotator\Rotator;
+
+class RotatableStreamDestination extends StreamDestination
+{
+    /**
+     * @var null|Rotator
+     */
+    private $preRotator;
+
+    /**
+     * @var null|Rotator
+     */
+    private $postRotator;
+
+    /**
+     * A constructor
+     *
+     * @param string $directory Path to backup directory.
+     * @param Rotator|null $preRotator Rotator to execute prior to push.
+     * @param Rotator|null $postRotator Rotator to execute after push.
+     */
+    public function __construct($directory, Rotator $preRotator = null, Rotator $postRotator = null)
+    {
+        parent::__construct($directory);
+        $this->preRotator = $preRotator;
+        $this->postRotator = $postRotator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function push($filename, LoggerInterface $logger)
```

Comment:
Maybe what we need is another interface which extends `Destination` that adds methods like list/remove and it could be Countable/IteratorAggregate. Something like:

```php
interface DestinationExtra extends Destination, \Countable, \IteratorAggregate
{
    /**
     * @return Backup[]
     */
    public function list();

    public function remove(Backup $backup);
}
```

Can't think of a good name for the class, `DestinationExtra` is not great.